docs/src/content/docs/getting-started/usage.mdx

```diff
@@ -0,0 +1,200 @@
+---
+
+import { Tabs, TabItem, Code, Aside, LinkCard } from '@astrojs/starlight/components';
+
+The simplest way to use `@sheepdog/svelte` is the following
+
+<Tabs syncKey="examples">
+	<TabItem label="Dot notation">
+	```svelte
+	<script>
+		import { task, timeout } from '@sheepdog/svelte';
+
+    	const myTask = task(async () => {
+    		// async stuff
+    	});
+    </script>
+
+    <button on:click={()=>{
+    	myTask.perform();
+    }}>
+    	do async stuff
+    </button>
+    ```
+    </TabItem>
+    <TabItem label="Option notation">
+    ```svelte
+    <script>
+    	import { task, timeout } from '@sheepdog/svelte';
+
+    	const myTask = task(async () => {
+    		// async stuff
+    	}, { kind: "default" });
+    </script>
+
+    <button on:click={()=>{
+    	myTask.perform();
+    }}>
+    	do async stuff
+    </button>
+    ```
+    </TabItem>
+
+</Tabs>
+this will run your async function when the user press the button. *"But why not use a normal async function?"* i hear you ask.
+
+**Because the `myTask` variable that you used to call `perform` is also a svelte store that contains
+a lot of useful information's about your task!**
+
+## `Task` structure
+
+- `isRunning`: whether the task is currently running or not
+- `last`: the last task instance, regardless of whether it errored, was canceled, or was successful
+- `lastCanceled`: the last canceled task instance
+- `lastErrored`: the last errored task instance
+- `lastRunning`: the last running task instance, as soon as the task stops running, this will be
+  undefined
+- `lastSuccessful`: the last successful task instance
+- `performCount`: the number of times the task has been run,
+
+```ts
+type Task = {
+	/**
+	 * whether the task is currently running or not
+	 */
+	isRunning: boolean;
+	/**
+	 * the last task instance, regardless of whether it errored, was canceled, or was successful
+	 */
+	last?: TaskInstance;
+	/**
+	 * the last canceled task instance
+	 */
+	lastCanceled?: TaskInstance;
+	/**
+	 * the last errored task instance
+	 */
+	lastErrored?: TaskInstance;
+	/**
+	 * the last running task instance, as soon as the task stops running, this will be undefined
+	 */
+	lastRunning?: TaskInstance;
+	/**
+	 * the last successful task instance
+	 */
+	lastSuccessful?: TaskInstance;
+	/**
+	 * number of times the task has been run,
+	 */
+	performCount: number;
+};
+```
+
+## `TaskInstance` structure
+
+- `error`: if an error is thrown inside the task instance, it will be found here
+- `isCanceled`: whether the task instance was canceled
+- `isError`: whether the task instance throw an error before completing
+- `isRunning`: whether the task instance is currently running
+- `isSuccessful`: whether the task instance completed successfully
+- `value`: if the task instance completed successfully, this will be the return value
+
+```ts
+type TaskInstance<TReturn> = {
+	/**
+	 * If an error is thrown inside the task instance, it will be found here.
+	 */
+	error?: undefined;
```

Comment:
I'm not great with TS but I don't think this is right

Reply:
Yeah that should be `Error` will fix